packages/@o3r/design/builders/generate-style/helpers/style-renderer-options.ts

```diff
@@ -113,14 +116,25 @@ export const getStyleRendererOptions = (tokenVariableNameRenderer: TokenKeyRende
 
   /** Sorting strategy of variables based on selected language */
   const tokenListTransforms = ((language) => {
+    const customSorter: DesignTokenListTransform[] = [];
+    if (options.sortOrderPatternsFilePath) {
+      try {
+        const regExps = (JSON.parse(readFileSync(resolve(context.workspaceRoot, options.sortOrderPatternsFilePath), {encoding: 'utf8'})) as string[])
+          .map((item) => new RegExp(item.replace(/^\/(.*)\/$/, '$1')));
+        customSorter.push(getTokenSorterFromRegExpList(regExps));
+      } catch {
+        context.logger.warn(`The specified RegExp file ${options.sortOrderPatternsFilePath} is not reachable or not correctly formatted.`);
+        context.logger.warn(`The order list will be ignored`);
+      }
+    }
     switch (language) {
       case 'scss':
       case 'sass': {
-        return [getTokenSorterByName, getTokenSorterByRef];
+        return [getTokenSorterByName, ...customSorter, getTokenSorterByRef];
```

Comment:
for my understanding.
This is sorting the variables by name and then applies the custom filter and then the ref one, the ref one having the highest priority ?

Reply:
This is correct